src/main/java/org/apache/sysds/runtime/matrix/data/LibMatrixReorg.java

```diff
@@ -1754,14 +1754,14 @@ static void transposeRow( double[] a, double[] c, int aix, int cix, int n2, int
 		//initial pass to determine capacity (this helps to prevent
 		//sparse row reallocations and mem inefficiency w/ skew
 		int[] cnt = null;
-		if( in.sparse && in.clen <= 4096 ) { //16KB
+		// if( in.sparse && in.clen <= 4096 ) { //16KB
```

Comment:
ensure some maximum bound on the temporary memory requirements. Also, I will have a look into more efficient transpose operations for ultra-sparse matrices.